src/Interpreters/Context.h

```diff
@@ -842,14 +845,19 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
     QueryAccessInfoPtr getQueryAccessInfoPtr() const { return query_access_info; }
     void setQueryAccessInfo(QueryAccessInfoPtr other) { query_access_info = other; }
 
-    void addQueryAccessInfo(
-        const String & quoted_database_name,
-        const String & full_quoted_table_name,
-        const Names & column_names);
+    enum class TableAccessInfoType : uint8_t
```

Comment:
It is not clear what this enum is good for. Column, partition and skip_index seem unrelated concepts.